### PR TITLE
Allow flexible vertex formats, remove hard coded formats from device.rs.

### PR DESCRIPTION
**Pull request by glennw**

### CODE REVIEW

**Comment by glennw** [comment]
r? @kvark 

This shouldn't have any noticeable changes, however it's a much cleaner implementation. This will make it simpler in the future to do things like a smaller vertex format for glyphs, which should help a lot with CPU usage on real world pages.

**Comment by glennw** [comment]
Rebased

File: webrender/src/debug_render.rs [hunk 1]
```diff
@@ -3,14 +3,69 @@
...
 use internal_types::{ORTHO_NEAR_PLANE, ORTHO_FAR_PLANE, TextureSampler};
-use internal_types::{DebugFontVertex, DebugColorVertex, RenderTargetMode};
+use internal_types::RenderTargetMode;
 use std::f32;
 use api::{ColorU, ImageFormat, DeviceUintSize};
 
+const DESC_FONT: VertexDescriptor = VertexDescriptor {
+    vertex_attributes: &[
+        VertexAttribute { name: "aPosition", count: 2, kind: VertexAttributeKind::F32 },
+        VertexAttribute { name: "aColor", count: 4, kind: VertexAttributeKind::U8 },
```
Comment (kvark) [inline webrender/src/debug_render.rs]: We should clearly differentiate between normalized and non-normalized integer semantics. So, I suggest using something like `U8Norm` here.

File: webrender/src/device.rs [hunk 2]
```diff
@@ -144,146 +153,97 @@ pub trait FileWatcherHandler : Send {
...
+impl VertexDescriptor {
+    fn bind(&self,
+            gl: &gl::Gl,
+            main: VBOId,
+            instance: VBOId) {
+        main.bind(gl);
+
+        let vertex_stride = self.vertex_attributes
+                                .iter()
+                                .fold(0, |stride, attr| stride + attr.size_in_bytes());
```
Comment (kvark) [inline webrender/src/device.rs]: nit: could be `map(|attr| attr.size_in_bytes()).sum()`

**Comment by glennw** [comment]
Thanks! Comments addressed.

**Comment by kvark** [comment]
Awesome, thanks!
@bors-servo r+

**Pull request merged**